dao/src/main/java/org/thingsboard/server/dao/device/DeviceProfileServiceImpl.java

```diff
@@ -173,6 +175,15 @@ public DeviceProfile saveDeviceProfile(DeviceProfile deviceProfile, boolean doVa
         } else if (deviceProfile.getId() != null) {
             oldDeviceProfile = findDeviceProfileById(deviceProfile.getTenantId(), deviceProfile.getId(), false);
         }
+
+        if (deviceProfile.getTransportType().equals(DeviceTransportType.LWM2M) &&
+                ((Lwm2mDeviceProfileTransportConfiguration)deviceProfile.getProfileData().getTransportConfiguration())
+                        .getClientLwM2mSettings()
+                        .getDefaultObjectIDVer() == null) {
+            ((Lwm2mDeviceProfileTransportConfiguration)deviceProfile.getProfileData().getTransportConfiguration())
+                    .getClientLwM2mSettings()
+                    .setDefaultObjectIDVer(V1_0.toString());
+        }
```

Comment:
Move presetting default value to UI, or in the place you use it - compare to null and use 1.0

Reply:
Ok